### PR TITLE
[RQ-1045] fix: rules table content header

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/componentsV2/ContentHeader/ContentHeader.tsx [hunk 1]
```diff
@@ -1,31 +1,68 @@
...
+        {searchValue !== undefined ? (
+          <Input
+            allowClear
+            prefix={<SearchOutlined />}
+            placeholder="Search by rule name"
+            value={searchValue}
+            onChange={(e) => setSearchValue(e.target.value)}
+            className="search-input"
+          />
+        ) : null}
```
Comment (RuntimeTerror10) [inline app/src/componentsV2/ContentHeader/ContentHeader.tsx:52]: Can we give a support to enable search in table or not by passing a prop named `enableSearch` ?

File: app/src/componentsV2/ContentHeader/ContentHeader.tsx [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
+                <div className={`filter-btn ${key === activeFilter ? "active" : ""}`}>
+                  <Button key={key} onClick={onClick} type={key === activeFilter ? "primary" : "text"}>
+                    {label}
+                  </Button>
+                </div>
+              ))
+            : null}
+        </div>
+      </Row>
+    </div>
```
Comment (RuntimeTerror10) [inline app/src/componentsV2/ContentHeader/ContentHeader.tsx:66]: Can we use Antd's `Segmented` component for filters?

File: app/src/features/rules/components/RulesList/components/RulesListIndex/RulesListIndex.tsx [hunk 3]
```diff
@@ -1,39 +1,314 @@
...
+              label: (
+                <div className="label">
+                  <MdOutlinePushPin className="icon" />
+                  Pinned
+                  {pinnedRecords.length ? <Badge count={pinnedRecords.length} overflowCount={20} /> : null}
+                </div>
+              ),
+              onClick: () => {
+                setActiveFilter("pinned");
+                console.log("pinned");
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesListIndex/RulesListIndex.tsx:282]: remove logs

File: app/src/features/rules/components/RulesList/components/RulesListIndex/RulesListIndex.tsx [hunk 4]
```diff
@@ -1,39 +1,314 @@
...
 
         <ContentHeader
           title="My Rules"
           subtitle="Create and manage your rules from here"
-          actions={[<Button type="primary">New Rule</Button>]}
+          actions={contentHeaderActions}
+          searchValue={searchValue}
+          setSearchValue={setSearchValue}
+          activeFilter={activeFilter}
+          filters={[
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesListIndex/RulesListIndex.tsx:258]: Store this this array of filters in  a var maybe named like `headerFilters` and pass this to filters prop.

File: app/src/componentsV2/ContentHeader/ContentHeader.tsx (hunk 1 above)
Comment (rohanmathur91) [reply app/src/componentsV2/ContentHeader/ContentHeader.tsx:52]: We again need to pass `searchValue` along with `enableSearch` prop instead let the content header decide with the `searchValue` only if to show search or not.

**Pull request merged**